bciers/apps/administration/tests/components/operators/OperatorDataGrid.test.tsx

```diff
@@ -0,0 +1,98 @@
+import { useRouter, useSearchParams } from "@bciers/testConfig/mocks";
+import OperatorDataGrid from "apps/administration/app/components/operators/OperatorDataGrid";
+import React from "react";
+
+useRouter.mockReturnValue({
+  query: {},
+  replace: vi.fn(),
+});
+
+useSearchParams.mockReturnValue({
+  get: vi.fn(),
+});
+
+const mockResponse = {
+  rows: [
+    {
+      id: 1,
+      legal_name: "Operator 1 Legal Name",
+      business_structure: "Sole Proprietorship",
+      cra_business_number: "123456789",
+      bc_corporate_registry_number: "abc1234567",
+    },
+    {
+      id: 2,
+      legal_name: "Existing Operator 2 Legal Name",
+      business_structure: "General Partnership",
+      cra_business_number: "123456789",
+      bc_corporate_registry_number: "def1234567",
+    },
+    {
+      id: 3,
+      legal_name: "New Operator 3 Legal Name",
+      business_structure: "BC Corporation",
+      cra_business_number: "123456789",
+      bc_corporate_registry_number: "ghi1234567",
+    },
+  ],
+  row_count: 2,
+};
+
+describe("OperatorDataGrid component", () => {
+  beforeEach(async () => {
+    vi.clearAllMocks();
+  });
+
+  it("renders the OperatorsDataGrid grid for internal users", async () => {
+    render(<OperatorDataGrid initialData={mockResponse} />);
+
+    // correct headers
+    expect(
+      screen.getByRole("columnheader", { name: "Legal Name" }),
+    ).toBeVisible();
+    expect(
+      screen.queryByRole("columnheader", { name: "Business Structure" }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole("columnheader", { name: "CRA Business Number" }),
+    ).toBeVisible();
+    expect(
+      screen.getByRole("columnheader", {
+        name: "BC Corporate Registry Number",
+      }),
+    ).toBeVisible();
+    expect(screen.getByRole("columnheader", { name: "Actions" })).toBeVisible();
+    expect(screen.queryAllByPlaceholderText(/Search/i)).toHaveLength(4);
+
+    // Check data displays
+    expect(screen.getByText(/Operator 1 Legal Name/i)).toBeVisible();
+    expect(screen.getByText(/Sole Proprietorship/i)).toBeVisible();
+    expect(screen.getAllByText(/123456789/i)).toHaveLength(3);
+    expect(screen.getByText(/abc1234567/i)).toBeVisible();
+    expect(screen.getAllByRole("link", { name: /View Details/i })).toHaveLength(
+      3,
+    );
+  });
+
+  // for later when operator details page is created
+  // it("renders the correct url for the operation information link", async () => {
+  //   render(
+  //     <OperatorDataGrid initialData={mockResponse} />,
+  //   );
+
+  //   const operationInfoLinks = screen.getAllByRole("link", {
+  //     name: /View Operation/i,
+  //   });
+
+  //   expect(operationInfoLinks[0]).toHaveAttribute(
+  //     "href",
+  //     "operations/1?operations_title=Operation+1",
+  //   );
+  //   expect(operationInfoLinks[1]).toHaveAttribute(
+  //     "href",
+  //     "operations/2?operations_title=Operation+2",
+  //   );
+  // });
```

Comment:
I suggest removing this test as it will pollute the code and let the user who tackles the related ticket to handle that. This way we have a cleaner codebase.